python/lsst/ts/watcher/threshold_handler.py

```diff
@@ -213,17 +233,53 @@ def get_severity_reason(self, value, current_severity, source_descr):
             value=value,
             source_descr=source_descr,
         )
+
+        tai_now = self.get_current_tai()
         for severity, scaled_level in self.severity_scaled_level_dict.items():
-            if scaled_value > scaled_level:
-                return make_severity_reason(severity=severity, with_hysteresis=False)
-            elif (
-                current_severity == severity
-                and scaled_value > scaled_level - self.hysteresis
-            ):
-                return make_severity_reason(severity=severity, with_hysteresis=True)
+            if math.isclose(self.level_periods[severity], 0.0):
+                # Immediately trigger an alarm if a value exceeds a threshold.
+                if scaled_value > scaled_level:
+                    return make_severity_reason(
+                        severity=severity, with_hysteresis=False
+                    )
+                elif (
+                    current_severity == severity
+                    and scaled_value > scaled_level - self.hysteresis
+                ):
+                    return make_severity_reason(severity=severity, with_hysteresis=True)
+            else:
+                # Only trigger an alarm if the change in a value exceeds a
+                # threshold within the given time range.
+                self.value_dict[tai_now] = value
+                # Remove too old items.
+                self.value_dict = {
+                    time: temp
+                    for time, temp in self.value_dict.items()
+                    if tai_now - time < self.level_periods[severity]
+                }
+                # Get all temperatures.
+                temperatures = [temp for temp in self.value_dict.values()]
+                # Determine if the temp change is too high.
+                if max(temperatures) - min(temperatures) > scaled_level:
+                    return make_severity_reason(
+                        severity=severity, with_hysteresis=False
+                    )
 
         return NoneNoReason
 
+    def get_current_tai(self) -> float:
+        """Get the current TAI time [UNIX seconds].
+
+        This method is designed to be overridden in unit tests.
```

Comment:
you could use pytest.monkeypatch instead.

Reply:
Thanks for the tip. I'll keep that in mind for a future implementation.